test/CMakeLists.txt

```diff
@@ -1,214 +1,120 @@
-target_link_libraries(SparqlAntlrParserTest gmock_main parser sparqlExpressions ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(SparqlAntlrParserTest)
-
-add_executable(SerializerTest SerializerTest.cpp)
-target_link_libraries(SerializerTest gtest_main absl::flat_hash_map ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(SerializerTest)
-
-add_executable(ParametersTest ParametersTest.cpp)
-target_link_libraries(ParametersTest gtest_main absl::flat_hash_map  ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(ParametersTest)
-
-add_executable(ZstdCompressionTest ZstdCompressionTest.cpp)
-target_link_libraries(ZstdCompressionTest gtest_main absl::flat_hash_map zstd ${cmake_thread_libs_init})
-gtest_discover_tests(ZstdCompressionTest)
-
-add_executable(TaskQueueTest TaskQueueTest.cpp)
-target_link_libraries(TaskQueueTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(TaskQueueTest)
-
-add_executable(SetOfIntervalsTest SetOfIntervalsTest.cpp)
-target_link_libraries(SetOfIntervalsTest gtest_main sparqlExpressions ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(SetOfIntervalsTest)
-
-add_executable(TypeTraitsTest TypeTraitsTest.cpp)
-target_link_libraries(TypeTraitsTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(TypeTraitsTest)
-
-add_executable(SparqlExpressionTest SparqlExpressionTest.cpp)
-target_link_libraries(SparqlExpressionTest gtest_main sparqlExpressions index ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(SparqlExpressionTest)
-
-add_executable(OStringStreamTest OStringStreamTest.cpp)
-target_link_libraries(OStringStreamTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(OStringStreamTest)
-
-add_executable(StreamableBodyTest StreamableBodyTest.cpp)
-target_link_libraries(StreamableBodyTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(StreamableBodyTest)
-
-add_executable(StreamableGeneratorTest StreamableGeneratorTest.cpp)
-target_link_libraries(StreamableGeneratorTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(StreamableGeneratorTest)
-
-add_executable(AcceptHeaderTest AcceptHeaderTest.cpp)
-target_link_libraries(AcceptHeaderTest gtest_main mediaTypes httpParser ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(AcceptHeaderTest)
-
-add_executable(RdfEscapingTest RdfEscapingTest.cpp)
-target_link_libraries(RdfEscapingTest gtest_main parser ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(RdfEscapingTest)
-
-add_executable(CompactStringVectorTest CompactStringVectorTest.cpp)
-target_link_libraries(CompactStringVectorTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(CompactStringVectorTest)
-
-add_executable(SparqlDataTypesTest SparqlDataTypesTest.cpp)
-target_link_libraries(SparqlDataTypesTest gmock_main engine ${CMAKE_THREAD_LIBS_INIT})
-gtest_discover_tests(SparqlDataTypesTest)
+addAndLinkTest(UnionTest engine)
+
+addAndLinkTest(TokenTest parser re2 -pthread ${ICU_LIBRARIES})
+
+addAndLinkTest(TurtleParserTest parser absl::flat_hash_map re2 -pthread)
```

Comment:
`-pthread` seems to be a little bit odd, noticed this in a lot of places.
Do you know what's the difference between this and `${CMAKE_THREAD_LIBS_INIT}` aren't they supposed to represent the same thing?

Reply:
They are. And CMAKE_THREAD_LIBS_INIT also should work if you run on windows,
using a windows threading library which doesn't know of posix.